ci/konflux/generate_component_definitions.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+import yaml
+
+
+"""
+This script is used to configure a Konflux Application with component definitions.
+We have very many components, and clicking them one by one in the UI is too inefficient.
```

Comment:
when you say we have many components - you mean what? nbc controller, kf nbc controller and?

Reply:
zero, one, many!

Reply:
each has two pipelines, so it's one, two, three, many when counting pipelines